api/src/main/kotlin/team/backend/domain/quiz/service/QuizServiceImpl.kt

```diff
@@ -15,17 +16,43 @@ import team.backend.model.ChatRole
 import team.backend.domain.prompt.entity.PositionType
 import team.backend.domain.prompt.entity.PromptType
 import team.backend.domain.quiz.entity.Quiz
+import team.backend.domain.quiz.policy.RANDOM_COUNT
+import team.backend.domain.quiz.query.QuizQuery
+import kotlin.random.Random
 
 @Service
 class QuizServiceImpl(
     private val quizReader: QuizReader,
     private val promptReader: PromptReader,
     private val memberReader: MemberReader,
-
     private val memberQuizAnswerStore: MemberQuizAnswerStore,
-
     private val openAiClient: OpenAiClient
 ): QuizService {
+    override suspend fun random(authorization: String): QuizQuery.RandomResponse {
+        val member = memberReader.get(authorization.toLong())
+
+        val quizzes = quizReader.getQuizByIds(getRandomlyQuizIds(member))
+            .asSequence()
+            .map { QuizQuery.Base(it.question, it.level.name, it.getJobName(), it.getSubjectName()) }
+            .toList()
+
+        return QuizQuery.RandomResponse(quizzes)
+    }
+
+    private suspend fun getRandomlyQuizIds(member: Member): List<Long> {
+        val quizIds = getCachedQuizIdsThatCanBeSolvedByMember(member)
+
+        if (quizIds.size <= RANDOM_COUNT) {
+            return quizIds
+        }
+
+        return quizIds.shuffled().take(RANDOM_COUNT)
```

Comment:
Wouldn't it be better to shuffle and to limit when you bring it from db?
query : `ORDER BY RAND()`, `LIMIT`

Reply:
@mangchhe 

1. Not support Natvie Query in Kotlin JDSL: https://github.com/line/kotlin-jdsl/issues/46
2. RAND() function is coupled mysql vendor